### PR TITLE
armcord: wayland screen sharing support

### PR DESCRIPTION
**Pull request by dev-null-undefined**
###### Description of changes
Based on https://github.com/ArmCord/ArmCord/issues/305
Added support for screen sharing on Wayland.

I also tried the wrapper flags used in vanila discord configuration
https://github.com/NixOS/nixpkgs/blob/d7705c01ef0a39c8ef532d1033bace8845a07d35/pkgs/applications/networking/instant-messengers/discord/linux.nix#L102
But screen sharing using those did not work.
 
###### Things done
- added flags to the wrapper based on this issue https://github.com/ArmCord/ArmCord/issues/305
- added pipewire and wayland libs as dependency 

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dev-null-undefined** [comment]
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>armcord</li>
  </ul>
</details>

**Comment by dev-null-undefined** [comment]
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>40 packages built:</summary>
  <ul>
    <li>adapta-gtk-theme</li>
    <li>cinnamon.cinnamon-common</li>
    <li>cinnamon.cinnamon-control-center</li>
    <li>cinnamon.cinnamon-gsettings-overrides</li>
    <li>cinnamon.cinnamon-screensaver</li>
    <li>firewalld-gui</li>
    <li>gnome-browser-connector</li>
    <li>gnome.gnome-control-center</li>
    <li>gnome.gnome-initial-setup</li>
    <li>gnome.gnome-session</li>
    <li>gnome.gnome-shell</li>
    <li>gnome.gnome-terminal</li>
    <li>gnome.gnome-tweaks</li>
    <li>networkmanager-fortisslvpn (gnome.networkmanager-fortisslvpn)</li>
    <li>networkmanager-iodine (gnome.networkmanager-iodine)</li>
    <li>networkmanager-l2tp (gnome.networkmanager-l2tp)</li>
    <li>networkmanager-openconnect (gnome.networkmanager-openconnect)</li>
    <li>networkmanager-openvpn (gnome.networkmanager-openvpn)</li>
    <li>networkmanager-vpnc (gnome.networkmanager-vpnc)</li>
    <li>gnome.nixos-gsettings-overrides</li>
    <li>gnomeExtensions.easyScreenCast</li>
    <li>gnomeExtensions.gsconnect</li>
    <li>gnomeExtensions.system-monitor</li>
    <li>libnma</li>
    <li>libnma-gtk4</li>
    <li>mojave-gtk-theme</li>
    <li>networkmanager-sstp</li>
    <li>networkmanager_strongswan</li>
    <li>networkmanagerapplet</li>
    <li>pantheon.elementary-greeter</li>
    <li>pantheon.elementary-session-settings</li>
    <li>pantheon.switchboard-plug-network</li>
    <li>pantheon.switchboard-with-plugs</li>
    <li>pantheon.wingpanel-applications-menu</li>
    <li>pantheon.wingpanel-indicator-network</li>
    <li>pantheon.wingpanel-with-indicators</li>
    <li>phosh</li>
    <li>phosh-mobile-settings</li>
    <li>vimix-gtk-themes</li>
    <li>whitesur-gtk-theme</li>
  </ul>
</details>

**Comment by dev-null-undefined** [comment]
Result of `nixpkgs-review pr 211859` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>armcord</li>
  </ul>
</details>

**Comment by dev-null-undefined** [comment]
Force pushed to resolve conflicts with upstream

**Comment by dev-null-undefined** [comment]
Result of `nixpkgs-review pr 211859` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>armcord</li>
  </ul>
</details>

File: pkgs/applications/networking/instant-messengers/armcord/default.nix [hunk 1]
```diff
@@ -114,9 +118,9 @@ stdenv.mkDerivation rec {
     makeWrapper $out/opt/ArmCord/armcord $out/bin/armcord \
       "''${gappsWrapperArgs[@]}" \
       --prefix XDG_DATA_DIRS : "${gtk3}/share/gsettings-schemas/${gtk3.name}/" \
+      --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform=wayland --enable-features=UseOzonePlatform --enable-features=WebRTCPipeWireCapturer }}" \
```
Comment (dev-null-undefined) [inline pkgs/applications/networking/instant-messengers/armcord/default.nix:121]: space before `}}` is needed after the `wrapGAppsHook` was added I am not sure why but the space fixed it.

**Review by dev-null-undefined — COMMENTED** [review]
Even tho I do not understand why the space is needed, I do not think it should stop this PR from being merged since it does fix the screen sharing issue and does not break anything else, I would say merge this PR and start new PR/Issue about why is the space needed.

**Comment by dev-null-undefined** [comment]
Result of `nixpkgs-review pr 211859` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>armcord</li>
  </ul>
</details>

**Comment by wrmilling** [comment]
Result of `nixpkgs-review pr 211859` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>armcord</li>
  </ul>
</details>

**Comment by wrmilling** [comment]
Armcord appears to function as expected on X11 still with these changes, I see no reason not to move forward with them.

LGTM

**Review by wrmilling — APPROVED** [review]

**Pull request merged**